motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -1968,14 +1968,29 @@ int Ros_MotionServer_GetDhParameters(Controller* controller, SimpleMsg* replyMsg
 
 int Ros_MotionServer_SetSelectedTool(Controller* controller, SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
 {
+#if !(DX100 || FS100)
+	MP_SET_TOOL_NO_SEND_DATA setToolData;
+	MP_STD_RSP_DATA responseData;
+#endif
+
 	int groupNo = receiveMsg->body.selectTool.groupNo;
 	int tool = receiveMsg->body.selectTool.tool;
 
 	if (groupNo >= 0 && groupNo < controller->numRobot)
 	{	
 		if (tool >= MIN_VALID_TOOL_INDEX && tool <= MAX_VALID_TOOL_INDEX)
 		{
+			//set tool that will be used by motion API
 			controller->ctrlGroups[receiveMsg->body.selectTool.groupNo]->tool = tool;
+
+#if !(DX100 || FS100)
+			//set jogging tool on the pendant
+			setToolData.sRobotNo = controller->ctrlGroups[receiveMsg->body.selectTool.groupNo]->groupId;
+			setToolData.sToolNo = tool;
+			mpSetToolNo(&setToolData, &responseData);
+#endif
+
+			//We don't care if mpSetToolNo fails. It won't affect the actual motion.
 			Ros_SimpleMsg_MotionReply(receiveMsg, ROS_RESULT_SUCCESS, 0, replyMsg, groupNo);

```

Comment:
My thinking is that a set-tool command from ROS should be primarily concerned with how it affects motion being commanded from ROS. That part is going to be successful as long as the tool number is valid. 

We can try to also set the jogging tool. But, the process shouldn't be stopped if that doesn't work for some arbitrary reason.  If the user then goes to pick up the pendant, the active tool number is displayed in the top status bar. So they can see what the current selection is at any time.
![image](https://user-images.githubusercontent.com/4623811/145831790-2436b40e-b919-481f-8d41-8a61a5a6180d.png)

Reply:
I agree with Ted.  
Thought it might seem strange to some, the Yaskawa controller manages the jog tool and the motion tool independently.  This is true for normal (none-ROS) operation.  It is so that  if you set your tool to manually jog the robot and then run some job, after when you want to jog the robot again, the jogging tool is the same as the one that was used before running the job.

Reply:
Ok, if you both agree then let's do it like this.

Is "motion tool" official terminology? If it is, we could use that to give the service added in #436 a better name.

Reply:
So just to be clear: is this PR now changing the behaviour from "only setting the motion tool" to "setting the motion tool and the jogging tool"? If so, from what you write @EricMarcil, that doesn't sounds like it would be desirable.

Reply:
No, there is no "official" terminology for it.  Manuals just make reference to tool in either case.

Reply:
So just to be clear: is this PR now changing the behaviour from "only setting the motion tool" to "setting the motion tool and the jogging tool"? If so, from what you write @EricMarcil, that doesn't sounds like it would be desirable.

Reply:
Ok, from https://github.com/ros-industrial/motoman/pull/436#discussion_r768843669 it's clear we do want `mpSetToolNo(..)` here, so closing as *resolved*.